### ISSUE TITLE
Error when using district id for London, ON

### ISSUE THREAD

**Issue by smffraser** [mampfes/hacs_waste_collection_schedule#1094]
TL;DR The service for London, ON does not work when a district ID is used due to the ID being formatted to all uppercase.

For London, Ontario, the data comes from the recycle coach service. 

When the district ID is used for London, the service errors out due to the district ID being converted to all upper case. It is a case sensitive string. For London, the district ID is "CityofLondon". The service code for recycle coach takes this district ID and runs self._format_key() on it, converting it to "CITYOFLONDON". This results in a "This pattern is not exist in database" error from the API.

The error can be confirmed using the test data provided in the recycle coach service script and adding on the project id, district id, and zone id (which have been confirmed to be correct values provided the district ID stays as 'CityofLondon'):

Source("1065 Sunningdale Rd E","london","ontario","528", "CityofLondon", "zone-z547")

**Comment by 5ila5** [comment]
fixed by #1098

Will be part of release 1.42.0 or is available as master/git version now
